beets/util/artresizer.py

```diff
@@ -232,23 +243,33 @@ def _check_method():
 
 def get_im_version():
     """Return Image Magick version or None if it is unavailable
-    Try invoking ImageMagick's "convert".
+    Try invoking ImageMagick's "magick". If "magick" is unavailable,
+    as with older versions, fall back to "convert"
+
+    Our iterator `im_legacy` will be non-zero when the first command
+    fails, and will be returned in a tuple along with the version
     """
-    try:
-        out = util.command_output(['convert', '--version'])
-
-        if b'imagemagick' in out.lower():
-            pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
-            match = re.search(pattern, out)
-            if match:
-                return (int(match.group(1)),
-                        int(match.group(2)),
-                        int(match.group(3)))
-            return (0,)
-
-    except (subprocess.CalledProcessError, OSError) as exc:
-        log.debug(u'ImageMagick check `convert --version` failed: {}', exc)
-        return None
+    cmds = ('magick', 'convert')
+    for im_legacy, cmd in enumerate(cmds):
+
+        try:
+            out = util.command_output([cmd, '--version'])
+
+            if b'imagemagick' in out.lower():
+                pattern = br".+ (\d+)\.(\d+)\.(\d+).*"
+                match = re.search(pattern, out)
+                if match:
+                    return ((int(match.group(1)),
+                            int(match.group(2)),
+                            int(match.group(3))),
+                            bool(im_legacy)
+                            )
```

Comment:
For legibility, something like this might be better:

```py
vers = (int(...), ...)
return vers, legacy
```

Reply:
Added in 48be3a7